handler/aws.go

```diff
@@ -27,6 +27,11 @@ func init() {
 		host := fmt.Sprintf("execute-api.%s.amazonaws.com", region)
 		services[host] = endpoints.ResolvedEndpoint{URL: fmt.Sprintf("https://%s", host), SigningMethod: "v4", SigningRegion: region, SigningName: "execute-api"}
 	}
+	// Add elasticsearch endpoints
+	for region := range endpoints.AwsPartition().Regions() {
+		host := fmt.Sprintf("%s.es.amazonaws.com", region)
```

Comment:
this should be taken care of?  where are you getting this hostname pattern from?  take a look here https://docs.aws.amazon.com/general/latest/gr/rande.html

You may not need this

Reply:
https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-request-signing.html

the aws elasticsearch service requests are also based on sigv4, so I think it is valid to extend this proxy to cover this use case too? 
(i actually come to know the signerV4 from the es service..

Reply:
have you tried making a call to an elasticsearch service endpoint using the proxy as it is?

Reply:
sorry for the late reply, to use for es was my use case at the first place indeed
- to access aws es, i choose to use iam for access control
- aws es comes with front end, kibana, so i use this proxy as reverse proxy with my first pr(overwriting host) to access es
- the kibana will check the `kbn-version` header or it requires me to send ` kbn-xsrf` header, so I added the function to forward all headers to upstream
- you could create a es without vpc(for easier public access), and access policy as below to verify
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "AWS": [
          "<user arn>"
        ]
      },
      "Action": "es:*",
      "Resource": [
        "arn:aws:es:<region>:<account id>:domain/<es domain>/*",
      ]
    },
  ]
}